src/Auth0.AuthenticationApi/AuthenticationApiClient.cs

```diff
@@ -10,199 +9,185 @@
+    /// Full documentation on the Auth0 Authentication API is available at https://auth0.com/docs/auth-api
     /// </remarks>
     public class AuthenticationApiClient : IAuthenticationApiClient, IDisposable
     {
-        private readonly Uri _baseUri;
-        private readonly ApiConnection _apiConnection;
-        private bool disposed = false;
+        readonly Uri baseUri;
+        readonly Uri tokenUri;
+        readonly IAuthenticationConnection connection;
+        bool disposed = false;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="AuthenticationApiClient" /> class.
         /// </summary>
         /// <param name="baseUri">Your Auth0 domain URI, e.g. https://tenant.auth0.com</param>
-        /// <param name="handler">The <see cref="HttpMessageHandler"/> which is used for HTTP requests.</param>
-        public AuthenticationApiClient(Uri baseUri, HttpMessageHandler handler = null)
-        {
-            _baseUri = baseUri;
-            _apiConnection = new ApiConnection(null, baseUri.AbsoluteUri, handler);
-        }
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="AuthenticationApiClient" /> class.
-        /// </summary>
-        /// <param name="baseUri">Your Auth0 domain URI, e.g. https://tenant.auth0.com</param>
-        /// <param name="httpClient">The <see cref="HttpClient"/> which is used for HTTP requests.</param>
-        public AuthenticationApiClient(Uri baseUri, HttpClient httpClient)
-        {
-            _baseUri = baseUri;
-            _apiConnection = new ApiConnection(null, baseUri.AbsoluteUri, httpClient);
-        }
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="AuthenticationApiClient" /> class.
-        /// </summary>
-        /// <param name="domain">Your Auth0 domain, e.g. tenant.auth0.com.</param>
-        /// <param name="handler">The <see cref="HttpMessageHandler"/> which is used for HTTP requests</param>
-        public AuthenticationApiClient(string domain, HttpMessageHandler handler = null)
-            : this(new Uri($"https://{domain}"), handler)
+        /// <param name="handler">Optional <see cref="IAuthenticationConnection"/> used to influence connection behavior.
+        /// Defaults to a freshly created <see cref="HttpClientAuthenticationConnection"/> that uses a single <see cref="HttpClient"/>.</param>
+        /// <remarks>To use a custom <see cref="HttpClient"/> or <see cref="HttpMessageHandler"/> create a 
+        /// <see cref="HttpClientAuthenticationConnection"/> passing that into the constructor. e.g.
+        /// <code>var client = new AuthenticationApiClient(new HttpClientAuthenticationConnection(myHttpClient));</code> or
+        /// <code>var client = new AuthenticationApiClient(new HttpClientAuthenticationConnection(myHttpMessageHandler));</code> or
+        /// </remarks>
+        public AuthenticationApiClient(Uri baseUri, IAuthenticationConnection connection = null)
```

Comment:
Why is this using `baseUri` and the next one is using `domain`?

Reply:
So you can either pass in a whole url, e.g.

 ```csharp
var x = new AuthenticationApiClient("https://damieng.auth0.com")
```

or just your domain name

```csharp
var x = new AuthenticationApiClient("damieng.auth0.com");
```

It was already part of the existing design and I didn't have a problem with it.

Reply:
Ok, I understand this better. It was unclear at first that it was only the first type that was different.